core/controllers/resources.py

```diff
@@ -77,10 +79,13 @@ def get(self, entity_type, entity_id, asset_type, encoded_filename):
             self.response.headers['Content-Type'] = str(
                 '%s/%s' % (asset_type, file_format))
 
-            if entity_type == feconf.ENTITY_TYPE_SUBTOPIC:
+            if page_context == feconf.ENTITY_TYPE_SUBTOPIC:
                 entity_type = feconf.ENTITY_TYPE_TOPIC
-                topic = topic_fetchers.get_topic_by_name(entity_id)
+                topic = topic_fetchers.get_topic_by_name(page_identifier)
                 entity_id = topic.id
+            else:
```

Comment:
Make this "elif page_context == one of the allowed entity types"

and add an "else throw InvalidInputException(...)" at the end

(and test these)

Reply:
Done